contrib/librdkafka-cmake/CMakeLists.txt

```diff
@@ -112,10 +116,19 @@ if(WITH_SASL_OAUTHBEARER)
   list(APPEND SRCS "${RDKAFKA_SOURCE_DIR}/rdkafka_sasl_oauthbearer.c")
 endif()
 
+## there is a conflict with the cJSON library from aws-s3
+# so normally we link cJSON lirary from contrib/aws (which have some extra patches to be thread safe)
+# but if we don't use aws-s3, we need to link the cJSON library from librdkafka (hacky)
+# which is (potentially) not thread safe (see https://github.com/confluentinc/librdkafka/issues/4159 ),
+if(NOT ENABLE_AWS_S3)
+   list(APPEND SRCS "${RDKAFKA_SOURCE_DIR}/cJSON.c")
```

Comment:
@filimonov the situation with one cJSON version having the fix and the other not is not really satisfying. Perhaps you would like to push a little bit here: https://github.com/confluentinc/librdkafka/issues/4159 ? Alternatively, what about a more workaround-y approach where a patched cJSON is maintained in contrib/librdkafka-cmake/ and copied by this CMakeList file into contrib/librdkafka/ ? Still not optimal but less fragile than having some (not all) vulnerable builds.

Reply:
Affected build are those which have aws disabled (freebsd i guess)...

Issue (data race) seems minor (even if that var will be damaged - nobody reads it). 

I would prefer to push https://github.com/confluentinc/librdkafka/issues/4159